### ISSUE TITLE
hashids.js produces different values than go/scala/.net library

### ISSUE THREAD

**Issue by nielsbergsma** [ivanakimov/hashids.js#23]
Hi,

It seems like hashids.js produces a different hash value than the libraries for other languages. I've written the test case below to test cross compatibility:

```javascript
var hashids = new Hashids("HelloSalt");
assert(hashids.encode(8612531504307815000) == "kAY4y6XzgDMeW");
```
The golang, scala and .net library all produce "kAY4y6XzgDMeW"

Niels

**Comment by ivanakimov** [comment]
Good catch @nielsbergsma.

I think it's just not handling big numbers properly, since `8612531504307815000` is much larger than `Number.MAX_SAFE_INTEGER`. I also checked it against 1.0.2 version, looks like it generated the same id there: `pmY4LkG2g56Ry` (which is still incorrect).

I'd probably have to throw an error or something similar, when it overflows like that. Otherwise, [we'd need to rewrite the algorithm](https://github.com/hashids/hashids.github.io/issues/2) to break down big numbers before encoding them (which would change ids for all implementations).

In the meantime a quick fix for you would be to split up the number into smaller numbers yourself before passing them to the libs.

**Comment by nielsbergsma** [comment]
Hi @ivanakimov,

Thanks for your response. Could you share an example how to split a big number of into smaller ones?

**Comment by ivanakimov** [comment]
You just gotta remember to add a `1` to the beginning of chunks in case they start with a `0`:

```javascript

var hashids = new Hashids('HelloSalt');

function encode(n) {
	var chunks = (n+'').match(/.{1,6}/g);
	for (var i = 0; i !== chunks.length; i++) {
		chunks[i] = '1'+chunks[i];
	}
	return hashids.encode.apply(hashids, chunks);
}

function decode(id) {
	var ret = '';
	var numbers = hashids.decode(id);
	for (var i = 0; i !== numbers.length; i++) {
		ret += (numbers[i]+'').slice(1);
	}
	return parseInt(ret, 10);
}

var id = encode(8612531504307815000); // QY48btxDEQfwZ2BsJ
var number = decode(id); // 8612531504307815000

console.log(id, number);
```

**Comment by tmorehouse** [comment]
The latest perl release uses `Math::BigInt,`, which allows to to handle extremely large integers >2^60, in order to get around older 32bit perl installs.

**Comment by niieani** [comment]
This should be fixed in 2.0 due to the BigInt support.
